File: Practica 3/Python/Pila.py
class Pila:
    def __init__(self, n):
        self.n = n
        self.arreglo = [0] * n
        self.top = -1

    def insert(self, e):
        if not self.isFull():
            self.top += 1
            self.arreglo[self.top] = e
        else:
            print(f"Pila llena. No se puede insertar {e}")

    def pop(self):
        if not self.isEmpty():
            elemento = self.arreglo[self.top]
            self.top -= 1
            return elemento
        else:
            print("Pila vacía. No se puede extraer elemento.")
            return None

    def peek(self):
        if not self.isEmpty():
            return self.arreglo[self.top]
        else:
            print("Pila vacía. No hay elemento en la cima.")
            return None

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.n - 1


# Ejemplo de uso
pila = Pila(5)
pila.insert(10)
pila.insert(20)
pila.insert(30)
print("Elemento en la cima:", pila.peek())
print("Elemento extraído:", pila.pop())
print("¿Está vacía?", pila.isEmpty())
pila.pop()
print("Tope de la pila después de pop:", pila.peek())

File: Practica 3/Java/Pila/src/PilaenJava/Pila.java
package PilaenJava;

public class Pila {
    private long[] arreglo;
    private int top;
    private int n;

    public Pila(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.top = -1;
    }

    public void insert(long e) {
        if (!isFull()) {
            arreglo[++top] = e;
        } else {
            System.out.println("Pila llena. No se puede insertar " + e);
        }
    }

    public long pop() {
        if (!isEmpty()) {
            return arreglo[top--];
        } else {
            System.out.println("Pila vacía. No se puede extraer elemento.");
            return -1;
        }
    }

    public long peek() {
        if (!isEmpty()) {
            return arreglo[top];
        } else {
            System.out.println("Pila vacía. No hay elemento en la cima.");
            return -1;
        }
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == n - 1;
    }
    public static void main(String[] args) {
        Pila pila = new Pila(5);
        pila.insert(10);
        pila.insert(20);
        pila.insert(30);
        System.out.println("Elemento en la cima: " + pila.peek());
        System.out.println("Elemento extraído: " + pila.pop());
        System.out.println("¿Está vacía? " + pila.isEmpty());
    }
}
